mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -556,6 +571,7 @@ class Word2VecModel private[spark] (
       .sortBy(- _._2)
       .take(num + 1)
       .tail
+      .map(v => (if (vecNorm == 0) v else (v._1, v._2 / vecNorm)))
```

Comment:
@srowen if either of the vector norms are 0, the cosine similarity should be 0. Though by definition I think here `v._2` will be 0 if the norm is 0, what do you think about making this `if vecNorm == 0) 0 ...` to be explicit?

(follows what you did in https://github.com/apache/spark/commit/94b39f7777ecff3794727c186bd681fa4c6af4fd#diff-88f4b62c382b26ef8e856b23f5167ccdR542)